pkg/apis/jenkins.io/v1/types_users.go

```diff
@@ -37,14 +37,15 @@ type UserList struct {
 
 // UserDetails containers details of a user
 type UserDetails struct {
-	Login             string             `json:"login,omitempty"  protobuf:"bytes,1,opt,name=login"`
-	Name              string             `json:"name,omitempty"  protobuf:"bytes,2,opt,name=name"`
-	Email             string             `json:"email,omitempty"  protobuf:"bytes,3,opt,name=email"`
-	CreationTimestamp *metav1.Time       `json:"creationTimestamp,omitempty" protobuf:"bytes,4,opt,name=creationTimestamp"`
-	URL               string             `json:"url,omitempty"  protobuf:"bytes,5,opt,name=url"`
-	AvatarURL         string             `json:"avatarUrl,omitempty"  protobuf:"bytes,6,opt,name=avatarUrl"`
-	ServiceAccount    string             `json:"serviceAccount,omitempty"  protobuf:"bytes,7,opt,name=serviceAccount"`
-	Accounts          []AccountReference `json:"accountReference,omitempty"  protobuf:"bytes,8,opt,name=accountReference"`
+	Login                string             `json:"login,omitempty"  protobuf:"bytes,1,opt,name=login"`
+	Name                 string             `json:"name,omitempty"  protobuf:"bytes,2,opt,name=name"`
+	Email                string             `json:"email,omitempty"  protobuf:"bytes,3,opt,name=email"`
+	CreationTimestamp    *metav1.Time       `json:"creationTimestamp,omitempty" protobuf:"bytes,4,opt,name=creationTimestamp"`
+	URL                  string             `json:"url,omitempty"  protobuf:"bytes,5,opt,name=url"`
+	AvatarURL            string             `json:"avatarUrl,omitempty"  protobuf:"bytes,6,opt,name=avatarUrl"`
+	ServiceAccount       string             `json:"serviceAccount,omitempty"  protobuf:"bytes,7,opt,name=serviceAccount"`
+	Accounts             []AccountReference `json:"accountReference,omitempty"  protobuf:"bytes,8,opt,name=accountReference"`
+	CreateServiceAccount bool               `json:"createServiceAccount,omitempty"  protobuf:"bytes,9,opt,name=createServiceAccount"`
```

Comment:
possibly `NativeUser` or something along those lines.

Reply:
> I _think_ I see what you mean: its a flag that is driving logic, which isn't quite right for CRDs. The bit I'm missing is: aren't many bits of CRDs state... that we then use to drive logic? e.g. "If this is a type _X_ User then do _this_, but if its a type _Y_ User then do _that_". This is why I was asking about the naming, would it make more sense if instead modelled as a `userType` field or something?

Yes something like that makes a lot more sense to me.

Reply:
👍 Great, thanks, I'll change the representation to something that indicates the type of User that was created.